### PR TITLE
Ospf_area_virtual_link

### PR DESCRIPTION
**Pull request by saichint**
This PR is for a new provider ospf_area_virtual_link.
All minitests pass on all nexus platforms.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Changelog
   * bfd (@saichint)
 * OSPF
   * ospf_area (@saichint)
+  * ospf_area_vl (@saichint)
```
Comment (chrisvanheuveln) [inline CHANGELOG.md]: I know the name is getting long. I'd prefer more than just an abbreviation though for customer-facing names. How does `ospf_area_vlink` sound? That's somewhat in line with our private-vlan / `pvlan` naming.

File: lib/cisco_node_utils/router_ospf_area_vl.rb [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+      hash = {}
+      RouterOspf.routers.each do |name, _obj|
+        # get all virtual_links under default vrf
+        links = config_get('ospf_area_vl', 'virtual_links', name: name)
+        unless links.nil?
+          hash[name] = {}
+          hash[name]['default'] = {}
+          links.each do |area, vl|
+            # combine area and virtual-link strings for uniqueness
+            avl = area + '_' + vl
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/router_ospf_area_vl.rb]: Blech. I would just nest the hash another level:

```
          links.each do |area, vl|
            hash[name]['default'][area] ||= {}
            hash[name]['default'][area][vl] = 
```

File: lib/cisco_node_utils/router_ospf_area_vl.rb [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      config_set('ospf_area_vl', 'message_digest_key_set', @set_args)
+    end
+
+    def dead_interval
+      config_get('ospf_area_vl', 'dead_interval', @get_args)
+    end
+
+    def dead_interval=(val)
+      set_args_keys(interval: val)
+      config_set('ospf_area_vl', 'dead_interval', @set_args)
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/router_ospf_area_vl.rb]: No `:state` value? How do you remove this setting or set it to default?

ditto for `hello_interval`, `retransmit_interval`, `transmit_delay`

File: tests/test_router_ospf_area_vl.rb [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+  def setup
+    super
+    remove_all_ospfs if @@pre_clean_needed
+    @@pre_clean_needed = false # rubocop:disable Style/ClassVars
+  end
+
+  def teardown
+    remove_all_ospfs
+    config 'no feature ospf'
```
Comment (chrisvanheuveln) [inline tests/test_router_ospf_area_vl.rb]: I don't think you need this feature disablement. It just slows down the test. If you really want it then I would make it part of the pre clean only.

**Comment by saichint** [comment]
Done with the review comment changes. All tests pass on all nexus

**Pull request merged**